plugins/task-plugin/src/task/che-task-provider.ts

```diff
@@ -53,6 +53,10 @@ export class CheTaskProvider {
             resultTarget.machineName = await this.machinePicker.pick();
         }
 
+        if (target && target.workingDir) {
+            resultTarget.workingDir = cheTaskDefinition.target.workingDir;
```

Comment:
You already have `target` here, so can take `workingDir` from `target` directly, like  `resultTarget.workingDir = target.workingDir`